src/rules/helpers.ts

```diff
@@ -0,0 +1,54 @@
+import type { RuleName } from './types';
+import type { MetaMaskRepository } from '../establish-metamask-repository';
+import type {
+  FailedPartialRuleExecutionResult,
+  PartialRuleExecutionResult,
+  Rule,
+  SuccessfulPartialRuleExecutionResult,
+} from '../execute-rules';
+
+/**
+ * Rule objects are fairly abstract: the name of a rule and the dependencies of
+ * a rule can be anything; and unfortunately, we cannot really enforce this or
```

Comment:
Sorry I'm having trouble following this sentence. It says that rules can be anything, but then that we can't enforce this. Enforce what? I thought rules could be anything.

Reply:
Er, you're right. Does this make more sense? [5989401](https://github.com/MetaMask/module-lint/pull/33/commits/59894010c62aab5f640da15a1b0f6402e9f61761)

Reply:
Not exactly; what does "enforce names" mean?

Reply:
Oh, I guess you mean that it ensures that other referenced rules exist. Ok, I think that makes sense.